### DISCUSSION TITLE
GitHub actions is failing due to "Error: Orchestrate failed - Error: ENOENT: no such file or directory, lstat"

### DISCUSSION THREAD

**Discussion by DineshChekuru** [community/community#52246] [Actions] [answered]
### Select Topic Area

General

### Body

Hi Team,

When I trying to deploy artifacts into Azure Synapse workspace I'm facing an "Error: Orchestrate failed - Error: ENOENT: no such file or directory, lstat". Below is my script.

name: workspace deployment

on:
    push:
        branches: [ workspace_publish ]
jobs:
    release:
        # You also can use the self-hosted runners.
           runs-on: windows-latest
           steps:
        # Checks out your repository under $GITHUB_WORKSPACE, so your job can access it.
           - uses: actions/checkout@v2
           - uses: azure/synapse-workspace-deployment@release-1.0
             with:
                               TargetWorkspaceName: 'syn098'
                               TemplateFile: ./syn098/TemplateForWorkspace.json
                               ParametersFile: ./syn098/TemplateParametersForWorkspace.json'
                               environment: 'Azure Public'
                               resourceGroup: 'synrg'
                               clientId: ${{secrets.CLIENTID}}
                               clientSecret:  ${{secrets.CLIENTSECRET}}
                               subscriptionId: ${{secrets.SUBSCRIPTIONID}}
                               tenantId: ${{secrets.TENANTID}}
                               DeleteArtifactsNotInTemplate: 'true'
                               managedIdentity: 'False'
                               
  As I'm using service principal to authenticate to Azure from GitHub, verified ClientID, ClientSecret, SubscriptionId, TenantId everything looks good but still facing the issue. As per my research I heard from discussion that we need to include "serviceAccountJsonPlainText" as variable in the script? If this is the issue I'm bit confused what secret we need to use in "serviceAccountJsonPlainText". Could you please let me know which value we need to use here?
  Please help me to resolve this issue

**Comment by SMelidoni** [comment]
Hi @DineshChekuru,

Based on the error message "Error: ENOENT: no such file or directory, lstat", it seems that the issue is related to the file path rather than the authentication. The `serviceAccountJsonPlainText` variable you mentioned is not applicable to Azure Synapse deployment.

Please ensure that the file paths for `TemplateFile` and `ParametersFile` are correct and the files exist in your repository. Make sure that the paths are relative to the root of your repository.

For example, if your files are located in the root of your repository, your paths should look like this: 
```
TemplateFile: ./TemplateForWorkspace.json
ParametersFile: ./TemplateParametersForWorkspace.json
```

Double-check that the paths are correct, and if necessary, update them in your script. If the issue persists, you may want to try running the script on a local machine to see if there are any issues with the files themselves or their content. Additionally, ensure that the files are committed and pushed to the `workspace_publish` branch, as that is the branch your workflow is listening to.

If none of these suggestions resolve the issue, please provide more information about your repository structure and the exact location of the template and parameters files.

**Comment by DineshChekuru** [comment]
Hi @SMelidoni I have re-checked the path of Template file and parameter file and I can confirm that path is correct which was checked in 'workspace_publish' branch only. I'm using the correct path in the script which you have mentioned ./TemplateForWorkspace.json,  ./TemplateParametersForWorkspace.json"

Please find the below script and paths of the templates for your reference.

name: workspace deployment

on:
    push:
        branches: [ workspace_publish ]
jobs:
    release:
        # You also can use the self-hosted runners.
           runs-on: windows-latest
           steps:
        # Checks out your repository under $GITHUB_WORKSPACE, so your job can access it.
           - uses: actions/checkout@v2
           - uses: azure/synapse-workspace-deployment@release-1.0
             with:
                               TargetWorkspaceName: 'snapse678'
                               TemplateFile: ./TemplateForWorkspace.json
                               ParametersFile: ./TemplateParametersForWorkspace.json'
                               #OverrideArmParameters: './path of the parameters.yaml'
                               environment: 'Azure Public'
                               CLIENTID : ${{secrets.CLIENTID}}
                               CLIENTSECRET : ${{secrets.CLIENTSECRET}}
                               resourceGroup: 'synrg'
                               subscriptionId: ${{secrets.SUBSCRIPTIONID}}
                               tenantId: ${{secrets.TENANTID}}
                               DeleteArtifactsNotInTemplate: 'true'
                               managedIdentity: 'False'

Templates path:
Repository : workspace_publish
Folder structure : repotestdel/syn098/ 
Files under the folder structure : [TemplateForWorkspace.json](https://github.com/DineshChekuru/repotestdel/blob/workspace_publish/syn098/TemplateForWorkspace.json)
[TemplateParametersForWorkspace.json](https://github.com/DineshChekuru/repotestdel/blob/workspace_publish/syn098/TemplateParametersForWorkspace.json)

Please let me know if we need to take any further action?

**Comment by SMelidoni** [comment]
Hi @DineshChekuru,

It appears that the file paths in your script do not match the folder structure you provided. In your script, you have:

```yaml
TemplateFile: ./TemplateForWorkspace.json
ParametersFile: ./TemplateParametersForWorkspace.json
```

But based on the folder structure you mentioned:

```yaml
Repository : workspace_publish
Folder structure : repotestdel/syn098/
Files under the folder structure : TemplateForWorkspace.json
TemplateParametersForWorkspace.json
```

You should update your script to reference the correct paths:
```yaml
TemplateFile: ./repotestdel/syn098/TemplateForWorkspace.json
ParametersFile: ./repotestdel/syn098/TemplateParametersForWorkspace.json
```

Make sure to update the paths in your script, commit the changes, and push to the workspace_publish branch. This should hopefully resolve the "Error: ENOENT: no such file or directory, lstat" error you're encountering.

If the issue still persists after updating the file paths, please provide any additional error messages or logs to help diagnose the problem further.

**Comment by DineshChekuru** [comment]
Hi @SMelidoni, as suggested tried changing the path in the script, still facing the same issue. 

Please find the error logs below for your reference.

with:
TargetWorkspaceName: 'snapse678'
TemplateFile: ./repotestdel/syn098/TemplateForWorkspace.json
ParametersFile: ./repotestdel/syn098/TemplateParametersForWorkspace.json
#OverrideArmParameters: './path of the parameters.yaml'
environment: 'Azure Public'
CLIENTID : ${{secrets.CLIENTID}}
CLIENTSECRET : ${{secrets.CLIENTSECRET}}
resourceGroup: 'synrg'
subscriptionId: ${{secrets.SUBSCRIPTIONID}}
tenantId: ${{secrets.TENANTID}}
DeleteArtifactsNotInTemplate: 'true'
managedIdentity: 'False'
Action failed -> Error: Orchestrate failed - Error: ENOENT: no such file or directory, lstat './repotestdel/syn098/TemplateParametersForWorkspace.json'

**Reply by SMelidoni** [reply]
Hi @DineshChekuru,

It seems that the issue is still related to the file path. The error message indicates that the file `./repotestdel/syn098/TemplateParametersForWorkspace.json` cannot be found. 

Please try the following steps to help identify the issue:

1. Double-check the file paths in your script and ensure they match the repository structure. Pay close attention to the case of the folder and file names, as the file system is case-sensitive.
2. Ensure that the `repotestdel/syn098/TemplateForWorkspace.json` and `repotestdel/syn098/TemplateParametersForWorkspace.json` files are committed and pushed to the `workspace_publish` branch.
3. Try using an absolute path for the template and parameters files. To do this, update your script to use the `$GITHUB_WORKSPACE` environment variable, which represents the root of your repository:

```yaml
TemplateFile: ${{ github.workspace }}/repotestdel/syn098/TemplateForWorkspace.json
ParametersFile: ${{ github.workspace }}/repotestdel/syn098/TemplateParametersForWorkspace.json
```

If the issue persists after trying these steps, please provide more information about your repository structure, the contents of the workspace_publish branch, and any other relevant details. This will help in diagnosing the issue further.

**Reply by DineshChekuru** [reply, marked as answer]
Hi @SMelidoni After changing the step "actions/checkout@v2" from "actions/checkout@v3" issue got resolved.
Old:
steps:
- uses: actions/checkout@v2

New:
steps:
- uses: actions/checkout@v3
